### PR TITLE
Custom processing pipeline

### PR DESCRIPTION
**Pull request by jabolina**
Added a custom processing pipeline, where we can remove some dependencies (Groovy). Also directly parsing the Kafka event to our JSON object, so we can remove Jackson too, I think.

### CODE REVIEW

File: src/main/java/org/gingesnap/cdc/chain/EventProcessingChainFactory.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+         case JSON:
+            ColumnJsonTranslator jsonTranslator = new ColumnJsonTranslator(columnsToUse);
+            // TODO: hardcoded value here
+            keyTranslator = backend.prefixRuleName() ? new PrependJsonTranslator(jsonTranslator, backend.jsonRuleName(), name) : jsonTranslator;
+            break;
+         default:
+            throw new IllegalArgumentException("Key type: " + backend.keyType() + " not supported!");
+      }
+
+      CacheService cacheService = Arc.container().instance(CacheService.class).get();
```
Comment (ryanemerson) [inline src/main/java/org/gingesnap/cdc/chain/EventProcessingChainFactory.java:52]: Should we be using `Arc` to look this up here, or would it be better to pass the `CacheBackend` via `create` as it's a mandatory requirement?

**Comment by ryanemerson** [comment]
Requires a rebase after https://github.com/gingersnap-project/db-syncer/pull/30

**Comment by jabolina** [comment]
Rebased and applied both suggestions. Thanks!

**Comment by ryanemerson** [comment]
Thanks @jabolina

**Pull request merged**